build-locally.sh

```diff
@@ -0,0 +1,127 @@
+green=$(tput setaf 76)
+white=$(tput setaf 7)
+tan=$(tput setaf 202)
+blue=$(tput setaf 25)
+
+#-----------------------------------------------------------------------------------------                   
+#
+# Headers and Logging
+#
+#-----------------------------------------------------------------------------------------                   
+underline() { printf "${underline}${bold}%s${reset}\n" "$@"
+}
+h1() { printf "\n${underline}${bold}${blue}%s${reset}\n" "$@"
+}
+h2() { printf "\n${underline}${bold}${white}%s${reset}\n" "$@"
+}
+debug() { printf "${white}%s${reset}\n" "$@"
+}
+info() { printf "${white}➜ %s${reset}\n" "$@"
+}
+success() { printf "${green}✔ %s${reset}\n" "$@"
+}
+error() { printf "${red}✖ %s${reset}\n" "$@"
+}
+warn() { printf "${tan}➜ %s${reset}\n" "$@"
+}
+bold() { printf "${bold}%s${reset}\n" "$@"
+}
+note() { printf "\n${underline}${bold}${blue}Note:${reset} ${blue}%s${reset}\n" "$@"
+}
+
+#-----------------------------------------------------------------------------------------                   
+# Main logic.
+#-----------------------------------------------------------------------------------------                   
+project="gradle"
+source_dir="."
+h1 "Building ${project}"
+
+# Debug or not debug ? Override using the DEBUG flag.
+if [[ -z ${DEBUG} ]]; then
+    export DEBUG=0
+    # export DEBUG=1
+    info "DEBUG defaulting to ${DEBUG}."
+    info "Over-ride this variable if you wish. Valid values are 0 and 1."
+else
+    info "DEBUG set to ${DEBUG} by caller."
+fi
+
+# Over-rode SOURCE_MAVEN if you want to build from a different maven repo...
+if [[ -z ${SOURCE_MAVEN} ]]; then
+    export SOURCE_MAVEN=https://galasadev-cicsk8s.hursley.ibm.com/main/maven/obr/
+    info "SOURCE_MAVEN repo defaulting to ${SOURCE_MAVEN}."
+    info "Set this environment variable if you want to over-ride this value."
+else
+    info "SOURCE_MAVEN set to ${SOURCE_MAVEN} by caller."
+fi
+
+# Create a temporary dir.
+# Note: This bash 'spell' works in OSX and Linux.
+if [[ -z ${LOGS_DIR} ]]; then
+    export LOGS_DIR=$(mktemp -d 2>/dev/null || mktemp -d -t "galasa-logs")
+    info "Logs are stored in the ${LOGS_DIR} folder."
+    info "Over-ride this setting using the LOGS_DIR environment variable."
+else
+    mkdir -p ${LOGS_DIR} 2>&1 > /dev/null # Don't show output. We don't care if it already existed.
+    info "Logs are stored in the ${LOGS_DIR} folder."
+    info "Over-ridden by caller using the LOGS_DIR variable."
+fi
+
+info "Using source code at ${source_dir}"
+cd ${BASEDIR}/${source_dir}
+if [[ "${DEBUG}" == "1" ]]; then
+    OPTIONAL_DEBUG_FLAG="-debug"
+else
+    OPTIONAL_DEBUG_FLAG="-info"
+fi
+
+# auto plain rich or verbose
+CONSOLE_FLAG=--console=plain
+
+log_file=${LOGS_DIR}/${project}.txt
+info "Log will be placed at ${log_file}"
+
+gradle --no-daemon \
+${CONSOLE_FLAG} \
+-Dorg.gradle.java.home=${JAVA_HOME} \
+-PsourceMaven=${SOURCE_MAVEN} ${OPTIONAL_DEBUG_FLAG} \
+clean build publishToMavenLocal \
```

Comment:
I think it would be a good idea to give the option, perhaps using an environment variable, to do just 'gradle build publishToMavenLocal' to use the cache of any previous builds for speed. It does not matter so much in this repository as it is very small and doesn't take long to build anyway, but may be a thought for the larger repos, framework onwards.

Reply:
Unlikely that you'd want to build anything and not publish it to local maven repo surely ?
In the other scripts I added the --clean | --delta build types, so not always a complete build. Good enough ?

Reply:
Yes I still think push to local maven repo, just have an option to remove the clean task for speed. Yes happy with that if it's an option for the other repos, thank you.